src/loss.jl

```diff
@@ -243,8 +242,8 @@ where `p` is equal to `1/(1 + exp.(scores))`. See also `logistic`.
 function bce(x̂,x;average=true) 
     ε = eltype(x̂)(1e-12)
     p = 1 ./ (1 .+ exp.(-x̂))
-    x =	convert(typeof(x̂),x)
-    l = x .* log.(p .+ ε) .+ (1 .- x).*log.((1-ε) .- p)
+    x =	oftype(value(x̂),x)
+    l = x .* log.(min.(p,ε)) .+ (1 .- x).*log.(1 .- max.(p,1-ε))
```

Comment:
?```l = x .* log.(max.(p,ε)) .+ (1 .- x).*log.(1 .- min.(p,1-ε))```

Reply:
That's right, sorry about that.  I have a possibly stupid question: isn't logistic and bce doing pretty much the same thing (one with {-1,1} the other with {0,1}?).  If that is correct why do we need epsilons in one and not the other?  Can't we just convert one gold to the other?